core/main/src/state/metrics_state.rs

```diff
@@ -224,12 +435,12 @@ impl MetricsState {
             let mut context = state.metrics.context.write().unwrap();
             let account_session = state.session_state.get_account_session();
             if let Some(session) = account_session {
-                context.account_id = session.account_id;
-                context.device_id = session.device_id;
+                context.account_id = Some(session.account_id);
+                context.device_id = Some(session.device_id);
                 context.distribution_tenant_id = session.id;
             } else {
-                context.account_id = "no.account.set".to_string();
-                context.device_id = "no.device_id.set".to_string();
+                context.account_id = None;
```

Comment:
This looks much cleaner, but how do we detect when these values are missing (in logfiles, etc. ?)

Reply:
They're now optional in SIFT 2.0, so if the values are missing the fields won't be there at all in the payload.